biz.aQute.launcher/src/aQute/launcher/Launcher.java

```diff
@@ -121,6 +122,9 @@
 	private boolean							restart					= false;
 	private boolean							frameworkInited			= false;
 
+	private ServiceRegistration<?>			launcherServiceRegistraion;
+	private Hashtable<String, Object>		launcherServiceProperties;
```

Comment:
This should not be an instance variable. You should not reuse the Hashtable across service calls. Create a new Hashtable and populated it at each place you make a service call.

Reply:
@bjhargrave I'll change it, but I'm unsure for the reason for it. I know that there had been issues in the past, because of service props that contained certain kinds of lists. besides this, we use this pattern in felix and equinox for years and it works like a charm. Can you elaborate some more?

Reply:
It is not clear in the registerService method who owns the Dictionary after the call. So the safest course is to assume the Dictionary cannot be reused and is owned by the framework. I recall that Tom Watson encountered some code where they reused the Dictionary and encountered side effects (probably because of mutated values such as collections). So we should always be safe and not reuse the Dictionary and any values.